ext/fg/js/source.js

```diff
@@ -94,7 +94,13 @@ class TextSourceRange {
                 this.rangeStartOffset === other.rangeStartOffset
             );
         } else {
-            return this.range.compareBoundaryPoints(Range.START_TO_START, other.range) === 0;
+            try {
+                return this.range.compareBoundaryPoints(Range.START_TO_START, other.range) === 0;
+            } catch (e) {
+                // Potential WrongDocumentError due to the ranges being in different documents.
+                // This can happen with shadow DOMs.
+                return false;
+            }
```

Comment:
Should this check `e.name` to avoid silencing other errors?

Reply:
c4b992408b5b16b9d4a3a199cad66cf20d73d6a9